src/adapter/src/coord/read_policy.rs

```diff
@@ -250,12 +250,15 @@ impl crate::coord::Coordinator {
                     let TimelineState { read_holds, .. } =
                         self.ensure_timeline_state(&timeline).await;
                     for (time, id_bundle) in &new_read_holds.holds {
-                        id_bundles.insert(Some(time.clone()), id_bundle.clone());
+                        id_bundles
+                            .entry(Some(time.clone()))
+                            .or_default()
+                            .extend(&id_bundle);
                     }
                     read_holds.extend(new_read_holds);
                 }
                 TimelineContext::TimestampIndependent | TimelineContext::TimestampDependent => {
-                    id_bundles.insert(None, id_bundle);
+                    id_bundles.entry(None).or_default().extend(&id_bundle);
```

Comment:
This one is likely to have been encountered and means that we may not have been setting the read policy for some static or timestamp dependent views.